### PR TITLE
Use the new search in all application lists

### LINKED ISSUE #873: vTHL4: Full text search
Improve the search functionality by offering a full text search based on all information in an application. One search field. This possibly replaces the column based filters or they should be improved to suit the improved functionality (create new tasks if need updating and not just removal).

- [x] search from all content of the application
- [x] one search field
- [x] try implementing full content search within backend first
- [ ] ~~search handled applications checkbox~~

### LINKED ISSUE #1352: Full-text search from all applicants/members' name
Related to #873

### PR DESCRIPTION
**Pull request by luontola**
Closes #1352, closes #873

On top of #1483, diff: https://github.com/CSCfi/rems/compare/search-more...search-cleanup

### CODE REVIEW

File: resources/translations/fi.edn [hunk 1]
```diff
@@ -300,9 +300,8 @@
            :phase-pending "Vaihe tulossa:"
            :phase-rejected "Vaihe hylätty:"
            :rejected "Hylätty"}
-  :search {:close-search "Sulje haku"
-           :open-search "Avaa haku"
-           :search-parameters "Hakutermit"}
+  :search {:clear-search "Nollaa etsintä"
```
Comment (opqdonut) [inline resources/translations/fi.edn]: Tyhjennä haku?

File: resources/translations/fi.edn [hunk 2]
```diff
@@ -300,9 +300,8 @@
...
+           :search "Etsi"}
```
Comment (opqdonut) [inline resources/translations/fi.edn]: Hae?

File: src/cljs/rems/application_list.cljs [hunk 3]
```diff
@@ -95,10 +97,29 @@
...
+    [table/table application-table]))
+
+(defn- application-list-defaults []
+  (let [config @(rf/subscribe [:rems.config/config])
+        id-column (get config :application-id-column :id)]
+    {:visible-columns #{id-column :description :resource :applicant :state :created :submitted :last-activity :view}
+     :default-sort-column :created
+     :default-sort-order :desc}))
+
+(defn default-component [{:keys [applications empty-message hidden-columns] :as opts}]
```
Comment (opqdonut) [inline src/cljs/rems/application_list.cljs]: docs please

also, I'm not sure I understand the name... how about renaming `component` to `list` and `default-component` to `component` or something?

File: src/cljs/rems/search.cljs [hunk 4]
```diff
@@ -0,0 +1,82 @@
+(ns rems.search
```
Comment (opqdonut) [inline src/cljs/rems/search.cljs]: docs please

File: resources/translations/fi.edn (hunk 2 above)
Comment (luontola) [reply resources/translations/fi.edn]: That would confuse it with "Hae uusia resursseja"

Comment (opqdonut) [reply resources/translations/fi.edn]: fair enough

**Comment by opqdonut** [comment]
Also, please add a CHANGELOG.md line for this!

**Pull request merged**